### ISSUE TITLE
What should I pay attention to when training with a single GPU?

### ISSUE THREAD

**Issue by BayMaxBHL** [zhyever/Monocular-Depth-Estimation-Toolbox#64]
When I tried to train on a single GPU, it was hard to get the same results as you can with multiple Gpus.

I tried to expand epoch, but the revenue wasn't good.

**Comment by BayMaxBHL** [comment]
Taking DepthFormer as an example, rmse=0.43 at 22k iterations using a single sheet of 3090ti.

**Comment by BayMaxBHL** [comment]
![image](https://user-images.githubusercontent.com/34785944/200515318-db52bf0a-612e-459b-8b27-52bb05d37e70.png)

**Comment by BayMaxBHL** [comment]
Make sure that there is no problem with the environment and dataset, because when training SimIPU, the training effect can reach the corresponding standard.

**Comment by zhyever** [comment]
That could be difficult. To adopt an 8GPU-based config to a 1GPU environment, basically, you need to extend the iteration number and tune down the learning rate. I cannot ensure that the reproduction of DepthFormer on 1GPU.

The Swin-T version of DepthFormer is based on 2 3090s, so it could be easier for you to modify this light version.

**Comment by BayMaxBHL** [comment]
非常感谢您对我的回复！
我后续没有停止训练，还在迭代，不过Swin-T 版本用一个3090Ti，可以训练到0.417，和您的log差的不太多。非常棒的复现体验。

**Comment by BayMaxBHL** [comment]
![image](https://user-images.githubusercontent.com/34785944/200552098-13cd58d1-da74-43b7-9a02-85d1119c4dc9.png)

**Comment by BayMaxBHL** [comment]
其实我是想向您咨询一下多GPU联合训练，是不是除了改变迭代次数和学习率还需要缩小BatchSize？
感觉联合训练的策略，多个GPU分别取多个批次后平均总损失，还是多个GPU瓜分一个批次后叠加损失？
感觉这个会对训练结果影响比较大的样子。

**Comment by BayMaxBHL** [comment]
Thank you very much for your reply!

I didn't stop training and I'm still iterating, but the Swin-T version uses a 3090Ti and can be trained to 0.417, which is not too far off your log. Great recreating experience.

In fact, I want to ask you about multi-GPU joint training, is it necessary to reduce the BatchSize in addition to changing the number of iterations and learning rate?

Sensory joint training strategy, multiple Gpus take the average total loss after multiple batches, or multiple Gpus divide a batch and add the loss?

It feels like this will have a big impact on the training results.

**Comment by zhyever** [comment]
That's great, but I think 0.417 v.s. 0.399 is still a gap, and you can try to modify some configs to bridge it.

Actually, it is tricky to tune bs and lr. For me, I'd like to linear scale them. For example `lr=0.01, bs/gpu=2, gpu_num=1` = `lr=0.02, bs/gpu=4, gpu_num=1` = `lr=0.04, bs/gpu=4, gpu_num=2`. 

For your other question, I forget the real implementation in PyTorch, but I think you can find the answer in the official docs.
